app/Http/Controllers/CourseEnrollmentController.php

```diff
@@ -53,13 +55,19 @@ public function store($id)
      */
     public function destroy($id)
     {
-        $course = DB::transaction(function () use ($id) {
-            $course = Course::findOrFail($id);
-            Auth::student()->unenroll($course);
+        try {
+            $course = DB::transaction(function () use ($id) {
+                $course = Course::findOrFail($id);
+                Auth::student()->unenroll($course);
 
-            return $course;
-        });
-        flash("You have successfully deleted the enrollment in {$course->name}.")->success();
+                return $course;
+            });
+            flash("You have successfully deleted the enrollment in {$course->name}.")->success();
+        } catch (StudentIsNotEnrolledInCourseException $e) {
+            flash('You cannot only delete enrollments that you have enrolled.')->error();
+        } catch (EnrollmentCannotBeDeleted $e) {
+            flash('You cannot delete an enrollment that already has a shift.')->error();
```

Comment:
Can we leverage the message that is already on the exception instance?

This suggestion is for both `catch` blocks.